### PR TITLE
Bugfix/credential proxy sequencing

### PR DESCRIPTION
**Pull request by jstuczyn**
This change is [Reviewable](https://reviewable.io/reviews/nymtech/nym/5187)

### CODE REVIEW

File: nym-credential-proxy/nym-credential-proxy/src/storage/mod.rs [hunk 1]
```diff
@@ -40,9 +42,15 @@ impl VpnApiStorage {
         let opts = sqlx::sqlite::SqliteConnectOptions::new()
             .filename(database_path)
             .create_if_missing(true)
-            .disable_statement_logging();
+            .log_statements(LevelFilter::Trace)
```
Comment (octol) [inline nym-credential-proxy/nym-credential-proxy/src/storage/mod.rs:45]: Trace by default?

File: nym-credential-proxy/nym-credential-proxy/src/http/state/mod.rs [hunk 2]
```diff
@@ -674,6 +676,52 @@ pub(crate) struct ChainWritePermit<'a> {
     inner: RwLockWriteGuard<'a, DirectSigningHttpRpcNyxdClient>,
 }
 
+impl<'a> ChainWritePermit<'a> {
+    // pub(crate) async fn make_transaction<F, Fut, T, E>(self, tx: F) -> Result<T, VpnApiError>
```
Comment (octol) [inline nym-credential-proxy/nym-credential-proxy/src/http/state/mod.rs]: Commented out code?

File: nym-credential-proxy/nym-credential-proxy/src/http/router/api/v1/ticketbook/shares.rs [hunk 3]
```diff
@@ -100,50 +100,51 @@ pub(crate) async fn query_for_shares_by_id(
     let uuid = random_uuid();
 
     let span = span!(Level::INFO, "query shares by id", uuid = %uuid, share_id = %share_id);
```
Comment (octol) [inline nym-credential-proxy/nym-credential-proxy/src/http/router/api/v1/ticketbook/shares.rs:102]: style: I would create a separate inner function. But ymmv of course

File: nym-credential-proxy/nym-credential-proxy/src/http/router/api/v1/ticketbook/shares.rs [hunk 4]
```diff
@@ -100,50 +100,51 @@ pub(crate) async fn query_for_shares_by_id(
     let uuid = random_uuid();
```
Comment (octol) [inline nym-credential-proxy/nym-credential-proxy/src/http/router/api/v1/ticketbook/shares.rs:100]: Hot take: uuid are too long to print in tracing statement. I prefer to truncate at 8 characters

File: nym-credential-proxy/nym-credential-proxy/src/deposit_maker.rs [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        info!("{} deposits made in transaction: {tx_hash}", replies.len());
+
+        let contract_data = match execute_res.to_contract_data() {
+            Ok(contract_data) => contract_data,
+            Err(err) => {
+                // that one is tricky. deposits technically got made, but we somehow failed to parse response,
+                // in this case terminate the proxy with 0 exit code so it wouldn't get automatically restarted
+                // because it requires some serious MANUAL intervention
+                error!("CRITICAL FAILURE: failed to parse out deposit information from the contract transaction. either the chain got upgraded and the schema changed or the ecash contract got changed! terminating the process. it has to be inspected manually. error was: {err}");
+                process::exit(0)
```
Comment (octol) [inline nym-credential-proxy/nym-credential-proxy/src/deposit_maker.rs]: What about return exit code signalling error? 0 is success

File: nym-credential-proxy/nym-credential-proxy/src/deposit_maker.rs [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        }
+
+        for (reply_channel, response) in replies.into_iter().zip(contract_data) {
+            let response_index = response.message_index;
+            let deposit_id = match response.parse_singleton_u32_contract_data() {
+                Ok(deposit_id) => deposit_id,
+                Err(err) => {
+                    // another impossibility
+                    error!("CRITICAL FAILURE: failed to parse out deposit id out of the response at index {response_index}: {err}. either the chain got upgraded and the schema changed or the ecash contract got changed! terminating the process. it has to be inspected manually");
+                    process::exit(0)
```
Comment (octol) [inline nym-credential-proxy/nym-credential-proxy/src/deposit_maker.rs]: Same as other comment

**Review by octol — APPROVED** [review]

File: nym-credential-proxy/nym-credential-proxy/src/storage/mod.rs (hunk 1 above)
Comment (jstuczyn) [reply nym-credential-proxy/nym-credential-proxy/src/storage/mod.rs:45]: i don't think we have to be logging all db statements at higher severity than that : )

**Pull request merged**